Package.swift

```diff
@@ -129,19 +129,46 @@ func isDebuggingMain(_ isDebug: Bool) {
     }
 }
 
+func xcodeFolder() -> String {
+    (Executable("/usr/bin/xcode-select")("-p") ?? "").trimmed
+}
+
+func swiftCompilerVersion() -> String {
+    let versionLine = Executable(
+        "\(xcodeFolder().trimmed)/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend"
+    )("--version") ?? ""
+
+    let regex = try? NSRegularExpression(pattern: " \\d.\\d")
+    return regex?.firstMatch(in: versionLine, range: NSRange(location: 0, length: versionLine.count)).flatMap {
+        Range($0.range, in: versionLine)
+    }.flatMap {
+        String(versionLine[$0]).trimmed
+    } ?? ""
+}
+
 extension PackageDescription.Target {
     func installSwiftSyntaxParser() {
         linkerSettings = [linkerSetting]
     }
 
     private var linkerSetting: LinkerSetting {
-        guard let xcodeFolder = Executable("/usr/bin/xcode-select")("-p") else {
-            fatalError("Could not run `xcode-select -p`")
-        }
-
-        let toolchainFolder = "\(xcodeFolder.trimmed)/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx"
+        let toolchainFolder = "\(xcodeFolder().trimmed)/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx"
+        return .unsafeFlags([
+            "-rpath", toolchainFolder
+        ])
+    }
+}
 
-        return .unsafeFlags(["-rpath", toolchainFolder])
+extension Version {
+    static var swiftSyntaxTag: Self {
+        switch swiftCompilerVersion() {
+        case "5.6": return "0.50600.1"
```

Comment:
What about 5.7?

Reply:
is there an official version? I could not find it in here:
https://github.com/apple/swift-syntax#declare-swiftpm-dependency-with-release-tag

Reply:
Ah, probably not released yet. Try next week 😜